### ISSUE TITLE
Can MA0020 check the type of the delegate argument to `FirstOrDefault`?

### ISSUE THREAD

**Issue by YoshiRulz** [meziantou/Meziantou.Analyzer#344]
**Is your feature request related to a problem? Please describe.**
MA0020 flags all `FirstOrDefault` calls with a `List<T>` receiver, but the delegate passed to `List<T>.Find` must be `Predicate<T>` and this doesn't seem to be considered.

**Describe the solution you'd like**
I think a semantic type check can go here:
https://github.com/meziantou/Meziantou.Analyzer/blob/c7b2738caddb1015b4c022182c6a57e6fef12fc0/src/Meziantou.Analyzer/Rules/OptimizeLinqUsageAnalyzer.cs#L215-L217

**Describe alternatives you've considered**
There was only one call flagged in my codebase and I was able to replace `Func<T, bool>` with `Predicate<T>`. But if I wasn't, I would have had to mute the warning for that call.

**Comment by meziantou** [comment]
Can you provide a repro where MA0020 wrongly reports an issue?

**Comment by YoshiRulz** [comment]
```csharp
private static readonly List<string> Numbers = new(new[] { "one", "two", "three", "four" });

private static string NumberMatching(Func<string, bool> predicate)
	=> Numbers.FirstOrDefault(predicate);

public static void Example()
	=> Console.WriteLine(NumberMatching(static s => !s.Contains('o')));
```

**Comment by meziantou** [comment]
Using the next version, it will not report the diagnostic by default when a conversion is needed.
You can edit the `.editorconfig` file with

````
[*.cs]
MA0020.report_when_conversion_needed=true
````

In this case it will report and fix using 

````c#
Numbers.Find(new Predicate<string>(predicate));
````
